cpp/harvest/nacjd.cc

```diff
@@ -345,27 +378,66 @@ void ParseJSONAndWriteMARC(MARC::Writer * const title_writer) {
             }
         }
     }
-    LOG_INFO("Processed: " + std::to_string(no_total) + "entries. " + std::to_string(no_initial_date) + " w/o initial date, "
+    LOG_INFO("Processed: " + std::to_string(no_total) + " entries. " + std::to_string(no_initial_date) + " w/o initial date, "
              + std::to_string(no_title) + " w/o title, " + std::to_string(no_creators) + " w/o creator and " + std::to_string(no_license)
              + " w/o license.");
 }
 
 
-} // unnamed namespace
+void getAll(int argc, char **argv) {
+    // Parse args
+    if (argc != 4)
+        Usage();
 
+    auto marc_reader(MARC::Reader::Factory(argv[2]));
+    auto marc_writer(MARC::Writer::Factory(argv[3]));
 
-int Main(int argc, char *argv[]) {
-    if (argc < 3 or argc > 4)
-        Usage();
+    // parse marc_file (later phase_x) and store ids in set
+    std::set<std::string> parsed_marc_ids;
+    LOG_INFO("Extracting existing ICPSR ids from marc input...");
+    ExtractExistingIDsFromMarc(marc_reader.get(), &parsed_marc_ids);
+    LOG_INFO("Found: " + std::to_string(parsed_marc_ids.size()) + " records with ICPSR ids.");
 
-    unsigned count_rows(9000);
-    if (argc == 4) {
-        count_rows = std::stoi(argv[3]);
-    }
+    // Download possible new publications and store in json file
+    LOG_INFO("Extracting ICPSR ids from website...");
+    unsigned number_of_new_ids;
+    ExtractIDsFromWebsite(parsed_marc_ids, &number_of_new_ids, 9000);
+    LOG_INFO(std::to_string(number_of_new_ids) + " new ids collected from website.");
+
+    // parse json file and store relevant information in variables
+    // write marc_records via marc_writer to marc_file
+    LOG_INFO("Parsing intermediate json file and save to marc output...");
+    ParseJSONAndWriteMARC(marc_writer.get());
+    LOG_INFO("Finished.");
+}
+
+void getByID(int argc, char **argv) {
+    // Parse args
+    if (argc != 4)
+        Usage();
 
-    auto marc_reader(MARC::Reader::Factory(argv[1]));
     auto marc_writer(MARC::Writer::Factory(argv[2]));
 
+    const unsigned id = std::stoi(argv[3]);
+
+    ExtractOneIDFromWebsite(id);
```

Comment:
We should also pass argv[2] to this function. This can be used as destination path for the JSON file (instead of the hardcoded nacjd_JSON_path variable in that function.